tests/unit/components/NcRichText/autoLink.spec.js

```diff
@@ -129,17 +129,34 @@ describe('autoLink', () => {
 			})
 		})
 
+		it('should not get route from relative link with base /nextcloud/index.php/apps/test/foo', () => {
+			getBaseUrl.mockReturnValue('https://cloud.ltd/nextcloud')
+			getRootUrl.mockReturnValue('/nextcloud')
+			const router = new VueRouter({
+				mode: 'history',
+				base: '/nextcloud/index.php/apps/test',
+				routes: [
+					{ path: '/foo', name: 'foo' },
+				],
+			})
+
+			expect(getRoute(router, '/nextcloud/index.php/apps/test/foo')).toBe(null)
+		})
+
 		// getRoute doesn't have to guarantee Talk Desktop compatiblity, but checking just in case
 		describe('with Talk Desktop router - no router base and invalid getRootUrl', () => {
 			it.each([
-				['https://cloud.ltd/nextcloud/index.php/apps/spreed?callTo=alice'],
-				['https://cloud.ltd/nextcloud/index.php/call/abc123ef'],
-				['https://cloud.ltd/nextcloud/index.php/apps/files'],
-				['https://cloud.ltd/nextcloud/'],
-			])('should not get route for %s', (link) => {
-				// On Talk Desktop both Base and Root URL returns an absolute path because there is no location
+				['https://cloud.ltd/nextcloud/index.php/apps/spreed?callTo=alice', '/apps/spreed?callTo=alice'],
+				['https://cloud.ltd/nextcloud/index.php/call/abc123ef', '/call/abc123ef'],
+				['https://cloud.ltd/nextcloud/index.php/apps/files', null],
```

Comment:
I don't get, why links like this or `https://cloud.ltd/apps/spreed` shouldn't work, if that's an app root (empty screen for Talk, home directory for Files)? 

Even for Talk Desktop `"file://.. ../talk_window/index.html#/apps/spreed"` is possible to open

P.S. Looking at the VueRouter setup in the test, maybe I've said some stupid things :see_no_evil: 
But in that case a plain test like `['/apps/spreed', '/apps/spreed']` (without queries) would be helpful to understand it

Reply:
`getRoute` returns client route - VueRouter's route for a relative or absolute URL. The path that use can use as `RouteLocationRaw`, as `<RouterLink :to=` value.

Reply:
> I don't get, why links like this ... shouldn't work

`https://cloud.ltd/nextcloud/index.php/apps/files` should not work for talk router, because `/apps/files` doesn't match any 

> or `https://cloud.ltd/apps/spreed` shouldn't work

This should work, there is no test where `https://cloud.ltd/apps/spreed` results with `null` for Talk / Talk Desktop